### PR TITLE
chore: teamsfx folder structure refactor

### PR DESCRIPTION
**Pull request by frankqianms**
teamsfx/app.yml -> ./teamsapp.yml
teamsfx/.env.{env} -> teamsAppEnv/.env.{env}
teamsfx/settings.json --write into--> ./teamsapp.yml

### CODE REVIEW

File: packages/fx-core/src/core/middleware/projectMigratorV3.ts [hunk 1]
```diff
@@ -93,15 +92,16 @@ import { getLocalizedString } from "../../common/localizeUtils";
 const Constants = {
   vscodeProvisionBicepPath: "./templates/azure/provision.bicep",
   launchJsonPath: ".vscode/launch.json",
-  appYmlName: "app.yml",
-  appLocalYmlName: "app.local.yml",
+  appYmlName: "teamsapp.yml",
+  appLocalYmlName: "teamsapp.local.yml",
```
Comment (xzf0587) [inline packages/fx-core/src/core/middleware/projectMigratorV3.ts:96]: There is a file versionMetadata. It contains the above two constant. Just use them

File: packages/fx-core/src/core/middleware/projectMigratorV3.ts [hunk 2]
```diff
@@ -93,15 +92,16 @@ import { getLocalizedString } from "../../common/localizeUtils";
...
   tasksJsonPath: ".vscode/tasks.json",
   reportName: "migrationReport.md",
   envWriteOption: {
     // .env.{env} file might be already exist, use append mode (flag: a+)
     encoding: "utf8",
     flag: "a+",
   },
+  environmentFolder: "teamsAppEnv",
```
Comment (xzf0587) [inline packages/fx-core/src/core/middleware/projectMigratorV3.ts:104]: I think this constant can be added to versionMetadata too. 
And as the env folder can be changed, defaultEnvironmentFolder could be better.

File: packages/fx-core/src/core/middleware/projectMigratorV3.ts [hunk 3]
```diff
@@ -455,17 +436,21 @@ export async function configsMigration(context: MigrationContext): Promise<void>
           // get envName
           const envName = fileNamesArray[2];
           // create .env.{env} file if not exist
-          await context.fsEnsureDir(SettingsFolderName);
-          if (!(await context.fsPathExists(path.join(SettingsFolderName, ".env." + envName))))
-            await context.fsCreateFile(path.join(SettingsFolderName, ".env." + envName));
+          await context.fsEnsureDir(Constants.environmentFolder);
+          if (
+            !(await context.fsPathExists(path.join(Constants.environmentFolder, ".env." + envName)))
```
Comment (xzf0587) [inline packages/fx-core/src/core/middleware/projectMigratorV3.ts:441]: use a constant to replace "env".

Comment (frankqianms) [reply packages/fx-core/src/core/middleware/projectMigratorV3.ts]: changed

File: packages/fx-core/src/core/middleware/projectMigratorV3.ts (hunk 2 above)
Comment (frankqianms) [reply packages/fx-core/src/core/middleware/projectMigratorV3.ts]: moved

File: packages/fx-core/src/core/middleware/projectMigratorV3.ts (hunk 1 above)
Comment (frankqianms) [reply packages/fx-core/src/core/middleware/projectMigratorV3.ts]: replaced

**Review by xzf0587 — APPROVED** [review]

**Pull request merged**